reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java

```diff
@@ -652,6 +652,17 @@ public final HttpServer proxyProtocol(ProxyProtocolSupportType proxyProtocolSupp
 		return dup;
 	}
 
+	/**
+	 * Specifies whether HTTP should be redirected to HTTPS if SSL is enabled.
+	 *
+	 * @param redirectHttp2Https true enables redirecting HTTP to HTTPS if SSL is enabled.
+	 */
+	public final HttpServer redirectHttp2Https(boolean redirectHttp2Https) {
```

Comment:
I'm wondering whether this has to be a configuration on `HttpServer` level or a configuration on `SslProvider` level. Wdyt?

Reply:
I was thinking the same thing.

**Another solution**
It also reasonable for `redirectHttp2Https` to be a parameter of `secure(SslProvider sslProvider, boolean redirectHttp2Https=false)` because redirectHttp2Https can only work when ssl is enabled.

**The reason it can also be an independent config**
I added `redirectHttp2Https` as an independent config just for fluent API especially the `sslProvider` in `secure(SslProvider sslProvider, boolean redirectHttp2Https=false)`, is usually used like `secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder))` and it's a little long and isn't that fluent to add `redirectHttp2Https` as the second parameter of `secure` method.

Reply:
> And it seems not good to add redirectHttp2Https as a config of SslProvider because SslProvider is designed at TCP level.

Yes agree

> It's also reasonable for redirectHttp2Https to be a parameter of secure(SslProvider sslProvider, boolean redirectHttp2Https=false) because redirectHttp2Https can only work when ssl is enabled.
 

```
public final HttpServer secure(Consumer<? super SslProvider.SslContextSpec> sslProviderBuilder, boolean redirectHttp)
public final HttpServer secure(SslProvider sslProvider, boolean redirectHttp)
```

Even though that it is a little longer, it is clear what's the intend.
Wdyt?

Reply:
OK, it's reasonable and I will update.